### DISCUSSION TITLE
Prisma Many-to-Many relations with findMany (select)

### DISCUSSION THREAD

**Discussion by pirmax** [prisma/prisma#17460] [Q&A]
I am facing an error with Prisma, it does not recognize my request which seems quite simple to me. I need to use "select" to retrieve only certain fields from the table.

## Post model:

    model Post {
      id          String    @id @default(cuid())
      title       String
      createdAt   DateTime? @default(now())
      categories CategoriesOnPosts[]
      keywords   KeywordsOnPosts[]
    
      @@map("posts")
    }

## Category model:

    model Category {
      id          String    @id @default(cuid())
      name        String
      createdAt   DateTime? @default(now())

      posts     CategoriesOnPosts[]
    
      @@map("categories")
    }

## CategoriesOnPosts model:

    model CategoriesOnPosts {
      postId       String
      categoryId   String
    
      post       Post     @relation(fields: [postId], references: [id])
      category   Category @relation(fields: [categoryId], references: [id])
    
      @@id([postId, categoryId])
      @@map("categoriesPosts")
    }

## My Prisma query:

    export const getPosts = async () =>
        await prisma.post.findMany({
            select: {
                id: true,
                title: true,
                categories: {
                    select: {
                        name: true,
                        slug: true,
                    },
                },
                createdAt: true,
            },
            orderBy: [
                {
                    createdAt: 'desc',
                },
            ],
        });

I get the following error and I don't know how to fix it.

> Unknown field `categories` for select statement on model Post.
> Available options are listed in green.

**Comment by nurul3101** [comment]
Hey @pirmax 👋 

Based on your schema, if you want to get name and slug from categories then you would need to use this query

```typescript
await prisma.post.findMany({
    select: {
      id: true,
      title: true,
      categories: {
        select: {
          category: {
            select: {
              name: true,
              posts: true,
            },
          },
        },
      },
      createdAt: true,
    },
    orderBy: [
      {
        createdAt: "desc",
      },
    ],
  });
```

Can you try it and let me know?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
